lib/sort_by_length.rb

```diff
@@ -3,5 +3,13 @@
 # Time complexity: ?
 # Space complexity: ?
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  split_sentence = my_sentence.split(" ")
+
+  split_sentence.each_with_index do |word, index|
+    previous_word = split_sentence[index - 1]
+    if index == 0 
+    elsif word.length < previous_word.length
+      split_sentence[index - 1], split_sentence[index] = word, previous_word
+    end
```

Comment:
Where is your 2nd loop here?  This just seems to be doing 1 pass of Bubblesort.  It's passing the tests, but it's not doing sorting...